File: server/dto/player_resources_dto.py
from dataclasses import dataclass

from dto.base_dto import BaseDto
from dto.spell_dto import SpellDto


@dataclass
class PlayerResourcesDto(BaseDto):
    maxHP: int
    currentHP: int
    maxMP: int
    currentMP: int
    # WARNING : this value must be an empty list for the
    # player1 if sent to player2 and vice-versa
    spells: list[SpellDto]

File: client/src/dto/player/PlayerResourcesDto.ts
export interface PlayerResourcesDto {
    maxHP: number;
    currentHP: number;
    maxMP: number;
    currentMP: number;
    currentStamina: number;
    maxStamina: number;
}

export const undefinedPlayerResources: PlayerResourcesDto = {
    maxHP: 0,
    currentHP: 0,
    maxMP: 0,
    currentMP: 0,
    currentStamina: 0,
    maxStamina: 0,
};